goaci.go

```diff
@@ -38,45 +41,75 @@ func debug(i ...interface{}) {
 	}
 }
 
+type StringVector []string
+
+func (v *StringVector) String() string {
+	return "\"" + strings.Join(*v, "\" \"") + "\""
+}
+
+func (v *StringVector) Set(str string) error {
+	*v = append(*v, str)
+	return nil
+}
+
 func main() {
+	var (
+		execOpts            StringVector
+		goDefaultBinaryDesc string
+		goBinaryOpt         string
+		goPathOpt           string
+	)
+
+	// Find the go binary
+	gocmd, err := exec.LookPath("go")
+
+	flag.Var(&execOpts, "exec", "Parameters passed to app, can be used multiple times")
+	if err != nil {
+		goDefaultBinaryDesc = "Go binary to use (default: none found in $PATH, so it must be provided)"
+	} else {
+		goDefaultBinaryDesc = "Go binary to use (default: whatever go in $PATH)"
+	}
+	flag.StringVar(&goBinaryOpt, "go-binary", gocmd, goDefaultBinaryDesc)
+	flag.StringVar(&goPathOpt, "go-path", "", "Custom GOPATH (default: a temporary directory)")
+	flag.Parse()
 	if os.Getenv("GOPATH") != "" {
-		die("to avoid confusion GOPATH must not be set")
+		warn("GOPATH envvar is ignored, use --go-path=\"$GOPATH\" option instead")
```

Comment:
Ah, that is going to be bikeshedding I'm afraid. Matter of opinions I think.

I prefer to have just one way of configuring the behavior of the application, in this case - just some parameters. I prefer to reserve env vars to configure behavior of some graphical toolkit or library the application might use. That is - behavior unrelated to what application does.

An example - you might want to configure toolkit with env vars to write some debug info to stdout or to use different backend (X11 or Wayland), but you configure behavior of graphical application by passing a parameter telling which file to open/process/whatever.

And about GOROOT - is there any point in using `go 1.3` and specifying GOROOT pointing to root of `go 1.4`? I think that better idea is to point `goaci` to using specific `go` binary, which already knows that it should use its own GOROOT (via `go env`). Also, requiring specifying GOROOT is bothersome and shouldn't be needed. I don't have it set in my environment.